maintenance/measure_tile_performance/measure_tile_performance.py

```diff
@@ -15,24 +16,59 @@
 NAME = 'snowplow'
 HOST = 'redshift.analytics.gov.bc.ca'
 PORT = 5439
+user = os.environ['lookeruser_rs']
+passwd = os.environ['lookerpass_rs']
+connection_string = (
+    f"dbname='{NAME}' "
+    f"host='{HOST}' "
+    f"port='{PORT}' "
+    f"user='{user}' "
+    f"password={passwd}")
 
 # connects with redshift database using environment variables
 # set redshift cache to 'off' before running any looker API calls
 # run the API calls
 # set redshift cache back to 'on' after the queries are processed.
 # close the redshift connection
+
+def cache_off():
+ # set redshift cache to 'off'
+    conn = psycopg2.connect(dsn=connection_string)
+    with conn:
+         with conn.cursor() as curs:
+            try:
+                curs.execute("ALTER USER looker SET enable_result_cache_for_session TO off;")
+                print('redshift cache is off and connection is closed')
+            except Exception as err:
+                print(f'Exiting due to psycopg2 execution error: {err}')
+                print('redshift cache is on and connection is closed')
+                exit(1)
+
+def cache_on():
+ # set redshift cache back to 'on'
+    sleep_timer = 0
+    while True and sleep_timer <= 50: 
+        conn = psycopg2.connect(dsn=connection_string)
+        with conn:
+            with conn.cursor() as curs:
+                try:
+                    curs.execute("ALTER USER looker SET enable_result_cache_for_session TO on;")
+                    print('redshift cache is on and connection is closed')
+                except Exception as err:
+                    # exponential backoff loop to reconnect with database 5 times
```

Comment:
You've implemented a linear back-off rather than an exponential back-off here because your sleep timer grows linearly with the number of the retry:

```
retry = 1, sleep = (retry * 10) = 10s
retry = 2, sleep = 20s
...
retry = 5, sleep = 50s
```

exponential would be one where you calculate the sleep with the retry attempt as an exponent, like:

```
retry = 1, sleep = (10s^retry) = 10s
retry = 2, sleep = 100s
...
retry = 5, sleep = 100,000s
```

doing a linear back-off is fine for this case because you give connections long enough that they shouldn't interfere with one another, but the comment should reflect that this is linear not exponential.